src/server/scripts/Kalimdor/RagefireChasm/boss_adarogg.cpp

```diff
@@ -0,0 +1,223 @@
+#include "ScriptMgr.h"
+#include "ScriptedCreature.h"
+#include "TaskScheduler.h"
+#include "ragefire_chasm.h"
+
+enum Spells
+{
+    // Intro
+    SPELL_ADJULES_CHOW_TIME         = 120113,
+    SPELL_EAT_TROGG                 = 120064,
+
+    SPELL_INFERNO_CHARGE_CAST       = 119405,
+    SPELL_INFERNO_CHARGE_SUMMON     = 119297, // Serverside
+    SPELL_FIRE_BREATH               = 119420
+};
+
+enum Texts
+{
+    SAY_INFERNO_CHARGE              = 0,
+
+    // Intro
+    SAY_HOUNDMASTER_INTRO_0         = 0,
+    SAY_HOUNDMASTER_INTRO_1         = 1
+};
+
+enum Events
+{
+    EVENT_INFERNO_CHARGE            = 1,
+    EVENT_FIRE_BREATH               = 2
+};
+
+enum Actions
+{
+    ACTION_KILL_HOUNDMASTERS        = 1
+};
+
+enum Waypoints
+{
+    // Intro Waypoint
+    PATH_INTRO                      = 6140800,
+
+    POINT_EAT_1                     = 0,
+    POINT_EAT_2                     = 3
+};
+
+constexpr Position IntroPoints[2] =
+{
+    { -282.31488f, -53.24906f, -60.802902f },
+    { -281.1894f,  -54.73433f, -60.34256f }
+};
+
+// Areatrigger - 7904
+class at_adarogg_intro : public OnlyOnceAreaTriggerScript
+{
+public:
+    at_adarogg_intro() : OnlyOnceAreaTriggerScript("at_adarogg_intro") { }
+
+    bool TryHandleOnce(Player* player, AreaTriggerEntry const* /*areaTrigger*/) override
+    {
+        if (InstanceScript* instance = player->GetInstanceScript())
+            if (Creature* adarogg = instance->GetCreature(BOSS_ADAROGG))
+                adarogg->AI()->DoAction(ACTION_KILL_HOUNDMASTERS);
+
+        return true;
+    }
+};
+
+// 61408 - Adarogg
+struct boss_adarogg : public BossAI
+{
+    using BossAI::BossAI;
+
+    boss_adarogg(Creature* creature) : BossAI(creature, BOSS_ADAROGG), eatCounter(0) { }
+
+    void Reset() override
+    {
+        _Reset();
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        BossAI::JustEngagedWith(who);
+        events.ScheduleEvent(EVENT_INFERNO_CHARGE, 10s);
+        events.ScheduleEvent(EVENT_FIRE_BREATH, 20s);
+    }
+
+    void JustDied(Unit* /*killer*/) override
+    {
+        _JustDied();
+    }
+
+    void DoAction(int32 action) override
+    {
+        if (action == ACTION_KILL_HOUNDMASTERS)
+        {
+            Creature* corruptedHoundmaster1 = me->FindNearestCreatureWithOptions(70.0f, { .StringId = "npc_corrupted_houndmaster_1" });
+            Creature* corruptedHoundmaster2 = me->FindNearestCreatureWithOptions(70.0f, { .StringId = "npc_corrupted_houndmaster_2" });
+
+            if (!corruptedHoundmaster1 || !corruptedHoundmaster2)
+                return;
+
+            corruptedHoundmaster1->AI()->Talk(SAY_HOUNDMASTER_INTRO_0);
+            corruptedHoundmaster2->AI()->Talk(SAY_HOUNDMASTER_INTRO_1);
+
+            scheduler.Schedule(2s, [this](TaskContext task)
+            {
+                DoCast(SPELL_ADJULES_CHOW_TIME);
+                me->GetMotionMaster()->MovePoint(0, IntroPoints[0]);
+
+                task.Schedule(1s, [this](TaskContext task)
```

Comment:
movement based on time feels a bit weird here, should this movement happen after `IntroPoint[0]` from above is reached?

Reply:
on retail he cast the spell in movement, before reach the point